src/metabase/integrations/common.clj

```diff
@@ -12,21 +12,17 @@
 (defn sync-group-memberships!
   "Update the PermissionsGroups a User belongs to, adding or deleting membership entries as needed so that Users is
   only in `new-groups-or-ids`. Ignores special groups like `all-users`, and only touches groups with mappings set."
-  [user-or-id new-groups-or-ids mapped-groups-or-ids sync-admin-group?]
-  (let [included-group-ids (cond-> (set (map u/the-id mapped-groups-or-ids))
-                             sync-admin-group? (conj (u/the-id (group/admin))))
-        excluded-group-ids (cond-> #{(u/the-id (group/all-users))}
-                             (not sync-admin-group?) (conj (u/the-id (group/admin))))
+  [user-or-id new-groups-or-ids mapped-groups-or-ids]
+  (let [mapped-groups-ids  (set (map u/the-id mapped-groups-or-ids))
+        excluded-group-ids #{(u/the-id (group/all-users))}
```

Comment:
since there's only one excluded group, might be clearer to get rid of this binding and just change the select statement to use `:not=`

Reply:
Actually, it's also used when we add membership(see line 34), and also I think having a name for it makes the logic clearer.